Source/CarthageKit/Xcode.swift

```diff
@@ -1080,85 +1080,152 @@ public typealias BuildSchemeProducer = SignalProducer<TaskEvent<(ProjectLocator,
+	let dependencyURL = rawDependencyURL.URLByResolvingSymlinksInPath!
+	let dependencyCheckoutsURL = dependencyURL.URLByAppendingPathComponent(CarthageProjectCheckoutsPath, isDirectory: true).URLByResolvingSymlinksInPath!
 
-			if isSymlink as? Bool == true {
-				// Since this dependency is itself a symlink, we'll create an
-				// absolute link back to the project's Build folder.
+	return SignalProducer<Bool, CarthageError>.attempt {
+		do {
+			try NSFileManager.defaultManager().createDirectoryAtURL(dependencyCheckoutsURL, withIntermediateDirectories: true, attributes: nil)
+		} catch let error as NSError {
+			if !(error.domain == NSCocoaErrorDomain && error.code == NSFileWriteFileExistsError) {
+				return .Failure(.WriteFailed(dependencyCheckoutsURL, error))
+			}
+		}
+		return .Success(true)
+	}
+	.flatMap(.Merge) { _ -> SignalProducer<Bool, CarthageError> in
+
+		return NSFileManager.defaultManager().carthage_enumeratorAtURL(rootCheckoutsURL, includingPropertiesForKeys: [NSURLIsDirectoryKey], options: [.SkipsSubdirectoryDescendants, .SkipsHiddenFiles])
+			.attemptMap { _, checkoutURL -> Result<Bool, CarthageError> in
+				var isDirectory: AnyObject?
 				do {
-					try NSFileManager.defaultManager().createSymbolicLinkAtURL(dependencyBinariesURL, withDestinationURL: rootBinariesURL)
+					try checkoutURL.getResourceValue(&isDirectory, forKey: NSURLIsDirectoryKey)
+					if isDirectory == nil {
+						return .Failure(.ReadFailed(checkoutURL, nil))
+					}
 				} catch let error as NSError {
-					return .Failure(.WriteFailed(dependencyBinariesURL, error))
-				}
-			} else {
-				// The relative path to this dependency's Carthage/Build folder, from
-				// the root.
-				let dependencyBinariesRelativePath = (dependency.relativePath as NSString).stringByAppendingPathComponent(CarthageBinariesFolderPath)
-				let componentsForGettingTheHellOutOfThisRelativePath = Array(count: (dependencyBinariesRelativePath as NSString).pathComponents.count - 1, repeatedValue: "..")
-
-				// Directs a link from, e.g., /Carthage/Checkouts/ReactiveCocoa/Carthage/Build to /Carthage/Build
-				let linkDestinationPath = componentsForGettingTheHellOutOfThisRelativePath.reduce(CarthageBinariesFolderPath) { trailingPath, pathComponent in
-					return (pathComponent as NSString).stringByAppendingPathComponent(trailingPath)
+					return .Failure(.ReadFailed(checkoutURL, error))
 				}
 
-				do {
-					try NSFileManager.defaultManager().createSymbolicLinkAtPath(dependencyBinariesURL.path!, withDestinationPath: linkDestinationPath)
-				} catch let error as NSError {
-					return .Failure(.WriteFailed(dependencyBinariesURL, error))
+				if let directory = isDirectory?.boolValue where directory {
+					let dependencySubCheckoutURL = dependencyCheckoutsURL.URLByAppendingPathComponent(checkoutURL.lastPathComponent!)
+					let foo = (CarthageProjectCheckoutsPath as NSString).stringByAppendingPathComponent(checkoutURL.lastPathComponent!)
```

Comment:
oh just realized I left a debugging variable in here... I'll rename this to not be `foo` =)